lambda/lambda.ml

```diff
@@ -311,6 +311,7 @@ type primitive =
   | Pint_as_pointer of locality_mode
   (* Atomic operations *)
   | Patomic_load of {immediate_or_pointer : immediate_or_pointer}
+  | Patomic_set of {immediate_or_pointer : immediate_or_pointer}
```

Comment:
I think `store` would be a more consistent name than `set`, but doesn't matter much

Reply:
Yes, I was unsure about it but concluded that `Set` is a friend of `Compare_set` like `Exchange` and `Compare_exchange`, instead of going with `load/store` because of the way they are handled throughout.